pkg/storage/kube/build.go

```diff
@@ -34,7 +40,48 @@ func (s *store) GetBuild(id string) (*brigade.Build, error) {
 	return b, err
 }
 
-// Get creates a new project and writes it to storage.
+// DeleteBuild deletes a build.
+func (s *store) DeleteBuild(bid string, options storage.DeleteBuildOptions) error {
+	opts := meta.ListOptions{
+		LabelSelector: fmt.Sprintf(jobFilter, bid),
+	}
+	delOpts := meta.NewDeleteOptions(0)
+	pods, err := s.client.CoreV1().Pods(s.namespace).List(opts)
+	if err != nil {
+		return err
+	}
+	if options.SkipRunningBuilds {
+		for _, p := range pods.Items {
+			if p.Labels["component"] == "build" {
+				if p.Status.Phase == v1.PodRunning || p.Status.Phase == v1.PodPending {
+					return errors.New("skipping because build is still running")
+				}
+			}
+		}
+	}
+	for _, p := range pods.Items {
+		log.Printf("Deleting pod %q", p.Name)
+		if err := s.client.CoreV1().Pods(s.namespace).Delete(p.Name, delOpts); err != nil {
+			log.Printf("failed to delete job pod %s (continuing): %s", p.Name, err)
+		}
+	}
+
+	secrets, err := s.client.CoreV1().Secrets(s.namespace).List(opts)
+	if err != nil {
+		return err
+	}
+	for _, sec := range secrets.Items {
```

Comment:
If `skipRunningBuilds` is `true` or more generally, pods remain for a build after this is called, should we forestall build secret deletion?

Reply:
Created https://github.com/Azure/brigade/issues/696 to track